lib/ash_archival/resource/changes/archive_related.ex

```diff
@@ -3,15 +3,16 @@ defmodule AshArchival.Resource.Changes.ArchiveRelated do
   use Ash.Resource.Change
   require Ash.Query
 
-  def change(changeset, _, _) do
+  def change(changeset, _, context) do
     archive_related = AshArchival.Resource.Info.archive_related(changeset.resource)
 
     if Enum.empty?(archive_related) do
       changeset
     else
       Ash.Changeset.after_action(changeset, fn changeset, result ->
         # This is not optimized. We should do this with bulk queries, not resource actions.
-        loaded = changeset.api.load!(result, archive_related)
+        opts = context |> Keyword.new()
```

Comment:
We can use `Ash.context_to_opts/1` to get `opts` here. This makes sense to do, but you've annotated this as not ready for merge. Does this change not resolve the issue?